Bottom: f4c642314fdeb6b38126127200ba7e981f70e226
Top:    0b7eb8fdc50b62442aa49d2a11a79eeb2bfc57f4
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-03-23 00:13:14 +0000

Fix to ParseGroupString()


---

diff --git a/src/misc.c b/src/misc.c
index 37aaebb..2df8cd9 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -1101,7 +1101,14 @@ ParseGroupString (char *group_string, LayerGroupType *LayerGroup, int *LayerN)
           /* ignore white spaces and check for separator */
           while (*s && isspace ((int) *s))
             s++;
-          if (!*s || *s == ':')
+
+          /* Backup so the outer loop terminates on the NULL termination */
+          if (*s == '\0') {
+            s--;
+            break;
+          }
+
+          if (*s == ':')
             break;
         }
       LayerGroup->Number[group] = member;
